task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -159,8 +159,18 @@ def run(ti: RuntimeTaskInstance, log: Logger):
         # TODO next_method to support resuming from deferred
         # TODO: Get a real context object
         ti.task.execute({"task_instance": ti})  # type: ignore[attr-defined]
-    except TaskDeferred:
-        ...
+    except TaskDeferred as defer:
+        next_method = defer.method_name
+        timeout = defer.timeout
+        kw = defer.kwargs or {}
+        # handle classpath on the server side
+        msg = DeferTask(
+            kwargs=kw,
+            next_method=next_method,
+            timeout=timeout,
+        )
+        global SUPERVISOR_COMMS
+        SUPERVISOR_COMMS.send_request(msg=msg, log=log)
```

Comment:
We should add a test case for this i.e. when what msg is sent when `run` raises Deferred

Reply:
Sure, makes sense. Working on it

Reply:
Only this is pending as of now

Reply:
Added a tets case in this commit https://github.com/apache/airflow/pull/44241/commits/dc39226f6659c36db83322bfed75faba274ee32a